src/OpenLoco/src/Windows/BuildVehicle.cpp

```diff
@@ -452,11 +517,22 @@ namespace OpenLoco::Ui::Windows::BuildVehicle
             }
 
             const auto* company = CompanyManager::get(companyId);
-            if (!Config::get().displayLockedVehicles && !company->isVehicleIndexUnlocked(vehicleObjIndex))
+            if (!(showUnlockedVehicles && company->isVehicleIndexUnlocked(vehicleObjIndex)) || (showLockedVehicles && !company->isVehicleIndexUnlocked(vehicleObjIndex)))
```

Comment:
you've removed the only usage of Config::displayLockedVehicles (which is the display cheat), so then to answer my previous comment on the cheat, we should remove it and the config option fully

Reply:
The options mentioned earlier should still be guarded by the cheat though. If they're not, then that's a bug